src/alley/wp/alleyvate/features/class-twitter-embeds.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Class file for Twitter_Embeds
+ *
+ * (c) Alley <info@alley.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package wp-alleyvate
+ */
+
+namespace Alley\WP\Alleyvate\Features;
+
+use Alley\WP\Types\Feature;
+use WP_Error;
+use WpOrg\Requests\Transport\Fsockopen;
+
+/**
+ * Twitter_Embeds feature.
+ */
+final class Twitter_Embeds implements Feature {
+	/**
+	 * Array of attempts to catch 404 responses from Twitter.
+	 *
+	 * @var int[] $attempts
+	 */
+	private array $attempts = [];
+
+	/**
+	 * Boot the feature.
+	 */
+	public function boot(): void {
+		add_filter( 'oembed_providers', [ $this, 'add_twitter_oembed_provider' ] );
+		add_filter( 'http_response', [ $this, 'filter_twitter_oembed_404s' ], 10, 3 );
+		add_filter( 'alleyvate_twitter_embeds_404_backstop', [ $this, 'attempt_404_backstop' ], 10, 3 );
+	}
+
+	/**
+	 * Add Twitter oEmbed provider.
+	 *
+	 * @param array{string, boolean}[] $providers Array of oEmbed providers.
+	 * @return array{string, boolean}[]
+	 */
+	public function add_twitter_oembed_provider( array $providers ): array {
+		/* phpcs:disable WordPress.Arrays.MultipleStatementAlignment */
+		return array_merge(
+			$providers,
+			[
+				'#https?://(www\.)?x\.com/\w{1,15}/status(es)?/.*#i' => [ 'https://publish.twitter.com/oembed', true ],
+				'#https?://(www\.)?x\.com/\w{1,15}$#i'               => [ 'https://publish.twitter.com/oembed', true ],
+				'#https?://(www\.)?x\.com/\w{1,15}/likes$#i'         => [ 'https://publish.twitter.com/oembed', true ],
+				'#https?://(www\.)?x\.com/\w{1,15}/lists/.*#i'       => [ 'https://publish.twitter.com/oembed', true ],
+				'#https?://(www\.)?x\.com/\w{1,15}/timelines/.*#i'   => [ 'https://publish.twitter.com/oembed', true ],
+				'#https?://(www\.)?x\.com/i/moments/.*#i'            => [ 'https://publish.twitter.com/oembed', true ],
+			],
+		);
+		/* phpcs:enable WordPress.Arrays.MultipleStatementAlignment.DoubleArrowNotAligned */
+	}
+
+	/**
+	 * Attempt to catch 404 responses from Twitter.
+	 *
+	 * @param array<mixed> $response    HTTP response.
+	 * @param array<mixed> $parsed_args HTTP request arguments.
+	 * @param string       $url         URL of the HTTP request.
+	 * @return array<mixed>
+	 */
+	public function filter_twitter_oembed_404s( array $response, array $parsed_args, string $url ): array {
```

Comment:
This is not documented here https://developer.wordpress.org/reference/hooks/http_response/ but the code implies it can get a `WP_Error` too.

https://github.com/WordPress/wordpress-develop/blob/1c7e99e1cc2dd6ab4881045f2ce3d3a2bf43792c/src/wp-includes/class-wp-http.php#L436-L438

And based on the output of `attempt_404_backstop`, that seems to be the case.
```suggestion
	 * @return array<mixed>|WP_Error
	 */
	public function filter_twitter_oembed_404s( array $response, array $parsed_args, string $url ): array|WP_Error {
```